src/ladds/Simulation.cpp

```diff
@@ -150,7 +153,14 @@ std::tuple<size_t, std::shared_ptr<HDF5Writer>, std::shared_ptr<ConjuctionWriter
       hdf5Writer = std::make_shared<HDF5Writer>(checkpointPath, false, 0);
     }
     // ... or write to new HDF5 file ....
-    if (const auto hdf5FileName = config.get<std::string>("io/hdf5/fileName", ""); not hdf5FileName.empty()) {
+    if (const auto hdf5FileName =
+            [&]() {
+              int rank{};
+              autopas::AutoPas_MPI_Comm_rank(AUTOPAS_MPI_COMM_WORLD, &rank);
+              const auto f = config.get<std::string>("io/hdf5/fileName", "");
+              return f.empty() ? "" : f + "_rank_" + std::to_string(rank);
+            }();
+        not hdf5FileName.empty()) {
```

Comment:
This is one very hard to read if statement. maybe to do the condition / parts in a separate line above?

Reply:
tbh I would like to keep it that way since it limits everything to the scope where it needs to be.
Best I can offer is extracting the definition of the lambda function.

```suggestion
    auto gethdf5FileName = [&]() {
        int rank{};
        autopas::AutoPas_MPI_Comm_rank(AUTOPAS_MPI_COMM_WORLD, &rank);
        const auto f = config.get<std::string>("io/hdf5/fileName", "");
        return f.empty() ? "" : f + "_rank_" + std::to_string(rank);
    };
    if (const auto hdf5FileName = gethdf5FileName(); not hdf5FileName.empty()) {
```

But I'm not convinced that this helps with the readability because it does not really affect the vertical layout but the horizontal of the init-statement + condition.

Reply:
It's better :)